Feat: Added a resources filter

shaheer-2000 opened the pull request:
<!-- If your PR fixes an open issue, use `Closes #101` to link your PR with the issue. #101 stands for the issue number you are fixing -->

Closes #140 

## 🛠️ Fixes issue(s)

<!-- Remove this section if not applicable -->

<!-- Example: Closes #31 -->

## 👨‍💻 Change(s) proposed

<!-- List all the proposed changes in your PR -->
- Adds a filters section with toggleable `project` and `topic` chips, that filter fetched resources

## ✔️ Checklist (Check all the applicable boxes) <!-- Follow the below conventions to check the box -->

<!-- Mark all the applicable boxes. To mark the box as done follow the following conventions -->
<!--
[x] - Correct; marked as done
[ ] - Not correct; marked as **not** done
-->

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## 📄 Note to reviewers

<!-- Add notes to reviewers if applicable -->
The `search-suggestion` class works really well as a chip, could possibly rename it so it can be appropriately used for more versatile cases, such as this one.

## 📷 Screenshots
![resources filter chips](https://i.imgur.com/s1MFAEB.png)

shaheer-2000 committed:
Feat: Added a resources filter

Ngoakor12 commented on frontend/src/pages/Resources/Resources.js:
```diff
@@ -34,9 +40,31 @@ export default function Resources() {
               <section className="resource-list">
                 <SearchForm />
+                <div className="search-suggestions">
+                  <h4 class="header-description">Filters:</h4>
```
- Can you please use the `font-size: 14px` for the "Filters:" text
- Also make sure the it is center horizontally with the filter chips

Ngoakor12 commented on frontend/src/pages/Resources/Resources.js:
```diff
@@ -34,9 +40,31 @@ export default function Resources() {
+                  {filters.map((filter, filterIdx) => {
+                    return (
+                      <p
+                        key={filterIdx}
```
Rather use the installed nanoid package for the `key` prop instead of the `filterIdx`

Ngoakor12 commented on frontend/src/pages/Resources/Resources.js:
```diff
@@ -34,9 +40,31 @@ export default function Resources() {
                 {renderedResources && renderedResources.length ? (
                   <div className="resources-list">
-                    <ResourceList resources={renderedResources} />
+                    {resourceFilter === "all" ? (
+                      <ResourceList resources={renderedResources} />
+                    ) : (
+                      <ResourceList
+                        resources={renderedResources.filter(
+                          (r) => r.type === resourceFilter
+                        )}
+                      />
+                    )}
```
NIcely done :)

- can you move the filtering logic to a variable at the top and just pass that variable here. Maybe a name like `filteredResources` could work.
- while on the filtering logic, rename `r` to a descriptive varible name, `resource`/`renderedResource` should work fine :)

Ngoakor12 reviewed (changes requested):
Mostly good...

- Regarding your comment about reusing that chip component, you can create one and move it here `./components/Chip/Chip.js` and then replace both the filter chips and search suggestions with it, it's up to you, otherwise I'll create a separate issue for it.
- Can the filters also stick to the top of the page just like the search suggestion chips please?

shaheer-2000 committed:
Add  component + Refactor to resolve requested changes

shaheer-2000 commented:
Requested changes should be resolved now.

- Created a `Chip` component and replaced search suggestions and filters with it.
- Renamed `.search-suggestions` to `.chips-wrapper` and `.search-suggestion` to `.chip`
- `fontSize` for `Filters: ` text is explicitly set to `14px` now
- Extracted the `filteredResources` bit to its own variable, as requested
- Replaced `filterIdx` with `nanoid()` for `key` prop value
- Filters are now `"sticky"`, the same as `suggestions`, as requested

That should be all, unless I am forgetting something

Ngoakor12 reviewed (changes requested):
Looks good to go, thanks :)
